### PR TITLE
last_commit_field_update

### PR DESCRIPTION
**Pull request by pg20**

### CODE REVIEW

**Comment by DMips** [comment]
@pg20 we need to refactor it a little bit - split it to smaller methods and add a class for most of the logic which you put in this rake task :)

**Comment by pg20** [comment]
Is the rake working fine otherwise ? Ill split in smaller methods in a day or two.

**Comment by DMips** [comment]
@pg20 it works for me without any problems :)

**Comment by pg20** [comment]
@DMips Hi I have updated the code into functions. I have added 2 filed in env.template, but I havent entered those global variable in rake file as i was not sure how to do it  and test it. If possible can u do this small change. Else evryhting is working fine. All error cases are handles like link not found. 
Redirection of url when repo move is still not handled in the github_api. I checked one more. Same issue in other api too. So I have displayed such urls on console and can be done manually along with 404 not found.

**Comment by pg20** [comment]
https://github.com/piotrmurach/github/issues/252 one guy has given solution for redirect. Can i somehow use his code by locally building github api ?

File: lib/tasks/git_commits.rake [hunk 1]
```diff
@@ -0,0 +1,70 @@
+namespace :git_commits do
+  desc "TODO"
+  task update_last_commit: :environment do
+   @projects = Project.all
+    @projects.each do |project|
```
Comment (DMips) [inline lib/tasks/git_commits.rake]: hmm I would like to extract it to separate module - do you want me to do that?

File: .env.template [hunk 2]
```diff
@@ -4,3 +4,5 @@ S3_KEY =
 S3_SECRET =
 S3_BUCKET_NAME =
 ROLLBAR_ACCESS_TOKEN =
+client_id=
+:client_secret
```
Comment (DMips) [inline .env.template]: when it comes to env we use uppercase names and I would also add github in the name to know later for which api this client and secret is about so `GITHUB_CLIENT_ID=` and `GITHUB_CLIENT_SECRET` or just `GITHUB_SECRET`

File: lib/tasks/git_commits.rake (hunk 1 above)
Comment (pg20) [inline lib/tasks/git_commits.rake]: Separate module as for example?

**Pull request closed without merging**